### DISCUSSION TITLE
Got Unexpected token '.' on Cpanel hosting

### DISCUSSION THREAD

**Discussion by Naimur53** [vercel/next.js#38715] [Help] [answered]
After build my application i have start the application locally by npm run start and its was ok but when I host it on cpanel I get a error  on stderr.log which is " 
  children: data?.map((single)=>/*#__PURE__*/ (0,react_jsx_runtime__WEBPACK_IMPORTED_MODULE_0__.jsx)(_SingleCollection__WEBPACK_IMPORTED_MODULE_4__/* ["default"] */ .Z, {
                           ^

SyntaxError: Unexpected token '.'
"
I have used ssr also SWR for internal fatch. I have tried to host an empty project that was ok on Cpanel. it working on vercel hosting but not in Cpanel

**Comment by icyJoseph** [comment, marked as answer]
What Node.js version is CPanel running? Feels like it doesn't understand `?.` syntax, which is available after Node.js 14. I think CPanel supports Node 16, but perhaps you have to tell it which version to use.

**Reply by Naimur53** [reply]
I have select 12.22.2 version on cpanel

**Reply by icyJoseph** [reply]
That must be it then. Node 12 doesn't support optional chaining as far as I remember. Coule you try with Node 16?

**Reply by Naimur53** [reply]
It's highest version available on my cpanel.. What to do?

**Reply by icyJoseph** [reply]
Mmm first thing is to develop in the same version that's used on that environment and confirm the issue. 

I think nowadays in Next 12, to the minimum supported Node version is 12.22.0

If you are using Typescript, perhaps try to change the target to an earlier version. ES5 for example.

I'm not sure how to tell babel/swc the Node target version.

Alternatively, use a docker container, but I'm not sure if that's possible in CPanel.

Last but not least, I remember reading that CPanel supports Node 16, perhaps there's something your account/environment is missing?

**Reply by icyJoseph** [reply]
https://support.cpanel.net/hc/en-us/articles/360051876714-How-to-install-NodeJS

Does the above apply to your account/use case?

**Reply by Naimur53** [reply]
I will try to update the node js
